semantics/language/translation/typing/expr.k

```diff
@@ -199,10 +200,19 @@ module C-TYPING-EXPR
      rule typeof(_ ? T:PointerType : _:IntegerType => T)
      rule typeof(_ ? t(Mods:Set, pointerType(T:Type)) : t(Mods':Set, pointerType(T':Type)) => t(Mods Mods', pointerType(compositeType(T, T'))))
           requires areCompatible(stripQualifiers(T), stripQualifiers(T'))
+               andBool notBool (isVariablyModifiedType(T) orBool isVariablyModifiedType(T'))
      rule typeof(_ ? t(Mods1:Set, pointerType(t(Mods1':Set, S:SimpleType))) : t(Mods2:Set, pointerType(t(Mods2':Set, void))) => t(Mods1 Mods2, pointerType(t(Mods1' Mods2', void))))
           requires notBool isFunctionType(t(Mods1', S))
+               andBool notBool isVariablyModifiedType(t(Mods1', S))
      rule typeof(_ ? t(Mods1:Set, pointerType(t(Mods1':Set, void))) : t(Mods2:Set, pointerType(t(Mods2':Set, S:SimpleType))) => t(Mods1 Mods2, pointerType(t(Mods1' Mods2', void))))
           requires notBool isFunctionType(t(Mods2', S))
+               andBool notBool isVariablyModifiedType(t(Mods2', S))
+
+     rule (.K => UNDEF("TTE1",
```

Comment:
It's the only place in that file.

But it looks like technically (_ ? NULL : ((*) [x]) NULL) probably doesn't require taking the composite type according to 6.5.15:6 (and should just have the type of (*)[x]), although that's pretty subtle...